tests/api/test_commands.py

```diff
@@ -240,3 +241,12 @@ def test_build_lazy(model_meta, tmp_path):
     build(
         "pip", model_meta, target=str(tmp_path / "build"), package_name="lol"
     )
+
+
+def test_serve_no_signature(model):
+    meta = MlemModel.from_obj(model)
+    with pytest.raises(
+        MlemError,
+        match="Cannot create interface from model with uspecified signature. Please re-save it and provide sample_data argument",
+    ):
+        serve(meta, "fastapi")
```

Comment:
Looks good! I think we should also issue an Exception when you build a Docker Image or try to deploy model somewhere.

Reply:
I thought about it, but build may occur in environments without model requirements, which means there will be no access to model signature and we wont be able to check it

Reply:
Can't we just check the presence of the signature? Like `if signature is not None`? Even if we don't load the actual content.

Reply:
From the naive perspective: since it exists in `model.mlem`, some magic should do the trick 🤔

Reply:
Ok, we need to find some solution here, because from the user perspective it can be pretty irritating if MLEM allows to build/deploy a model, but later fails to serve it. Let's think about this some more time then.

Reply:
Btw, do I understand correctly, that if it's `UnspecifiedDataType`s, then we don't need any model requirements installed to local signature? Or we still need let's say `PyTorch` if model was a PyTorch one?
What I mean, is that if we try to deserialize signature and we can't because of missing requirements, then it's not `UnspecifiedDataType` and everything is ok.

But we probably we deserialize signature as part of deserializing a model itself, so that will fail before it reaches the signature?

Reply:
Yes, we need model requirements because data types are nested fields. Also, we need data type requirements, because theoretically only part of all data types can be unspecified

Reply:
> only part of all data types can be unspecified

If that happens to be, will `serve` fail?

Then it looks like the only alternative to checking raw objects is to try to check anyway - if we can't load some dependencies, we're skipping this, but just issue a warning about it.

Reply:
Good point! Then we need to check servers somehow 🤔 
What I'm advocating here for: we should fail as early on as possible. Otherwise it's going to be a bad experience for users :)

Reply:
I get it. But not all servers/builders require signature to be present. Eg venv builder
A bit ugly, but we can add classvar for servers  which will enable signature checking.
And we can check raw methods objects `type` field